### PR TITLE
Mview patch update

### PR DESCRIPTION
**Pull request by skovalenk**
### Description (*)
New Mview approach which will allow to do partial reindex for EAV entities

### Related Pull Requests
[SAAS-EXPORT] https://github.com/magento/saas-export/pull/117
[EE] https://github.com/magento/partners-magento2ee/pull/320/files

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/pull/29888#issue-478624466

### CODE REVIEW

File: lib/internal/Magento/Framework/Mview/View/ChangelogInterface.php [hunk 1]
```diff
@@ -11,6 +11,11 @@
  */
 interface ChangelogInterface
 {
+    const ATTRIBUTE_SCOPE_SUPPORT = 'attribute_scope';
+    const STORE_SCOPE_SUPPORT = 'store_scope';
+    const ATTRIBUTE_COLUMN = 'attribute_id';
+    const STORE_COLUMN = 'store_id';
```
Comment (kandy) [inline lib/internal/Magento/Framework/Mview/View/ChangelogInterface.php:17]: Better to name this more agnostic like const SCOPE_COLUMN = "scope_id";
Different attributes can have different scope (global, store, website level)

File: lib/internal/Magento/Framework/Mview/Config/Converter.php [hunk 2]
```diff
@@ -28,6 +28,8 @@ public function convert($source)
             $data['view_id'] = $viewId;
             $data['action_class'] = $this->getAttributeValue($viewNode, 'class');
             $data['group'] = $this->getAttributeValue($viewNode, 'group');
+            $data['store_scope'] = $this->getAttributeValue($viewNode, 'store_scope');
```
Comment (akaplya) [inline lib/internal/Magento/Framework/Mview/Config/Converter.php:31]: Explicit knowledge about the application entities such as a store, an attribute, should not be mentioned at the Framework level. By the way, these are not the only scopes available in Magento. Could you please make it more configurable, to make sure that adding of a new scope, like a shared catalog will not require code changes.

**Review by akaplya — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Mview/Config/Converter.php (hunk 2 above)
Comment (mslabko) [reply lib/internal/Magento/Framework/Mview/Config/Converter.php:31]: "shared catalog" relies on existing data. 
We can rename it to "scope" to be more abstract, but I do not see any reason to provide complex configuration which does not have any use-cases

**Comment by mslabko** [comment]
@magento run all tests

**Comment by skovalenk** [comment]
Closing this and moving this to saas-export repo

**Pull request closed without merging**